Fix SwiftLint errors

mokagio opened the pull request:
Builds on top of #12017 and addresses all the errors and some of the warnings SwiftLint picked up.

Recommend reviewing commit by commit.

mokagio committed:
Track result of running SwiftLint autocorrect

mokagio committed:
Address `discarded_notification_center_observer` SwiftLint violations

mokagio committed:
Address or silence SwiftLint `weak_delegate` violations

mokagio committed:
Address SwiftLint `overridden_super_call` violations

mokagio committed:
Address some `localization_comment` SwiftLint violations

There are more of those, but I'm running out of time to address them 😳

dangermattic commented:
<!--
  0 Errors
  2 Warnings: View files have been modified,...
  0 Messages
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          2 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">View files have been modified, but no screenshot or video is included in the pull request. Consider adding some for clarity.</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is larger than 500 lines of changes. Please consider splitting it into smaller PRs for easier and faster reviews.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

iangmaia commented on Experiments/Experiments/FeatureFlag.swift:
```diff
@@ -149,7 +149,7 @@ public enum FeatureFlag: Int {
     case shareProductAI
```
These errors are still being reported on CI:

- `WooCommerce/Classes/ViewModels/Order Details/OrderDetailsDataSource.swift:1365:52`: Localized strings should include a description giving context for how the string is used. (localization_comment)
- `WooCommerce/Classes/ViewRelated/Dashboard/Settings/CardReadersV2/SetUpTapToPayCompleteViewController.swift:135:32`: Localized strings should include a description giving context for how the string is used. (localization_comment)
- `WooCommerce/Classes/ViewRelated/Dashboard/Settings/CardReadersV2/SetUpTapToPayInformationViewController.swift:199:28`: Localized strings should include a description giving context for how the string is used. (localization_comment)
- `WooCommerce/Classes/ViewRelated/Search/SearchViewController.swift:341:21`: Localized strings should include a description giving context for how the string is used. (localization_comment)
- `Yosemite/Yosemite/Model/Mocks/Graphs/ScreenshotsObjectGraph.swift:13:26`: Localized strings should include a description giving context for how the string is used. (localization_comment)
- `Yosemite/Yosemite/Model/Mocks/Graphs/ScreenshotsObjectGraph.swift:14:31`: Localized strings should include a description giving context for how the string is used. (localization_comment)
- `Yosemite/Yosemite/Model/Mocks/Graphs/ScreenshotsObjectGraph.swift:15:31`: Localized strings should include a description giving context for how the string is used. (localization_comment)

mokagio replied:
🤔 interesting to see that CI considers them error when they are but warnings on my end.

mokagio replied:
Actually, not interesting at all, given CI uses `--strict` 😅

mokagio committed:
Fix access control issue in unit test

See how the CI build failed
https://buildkite.com/automattic/woocommerce-ios/builds/19633#018dac54-625d-4f97-8761-3fa649b48bec/2644-4487

Shame on me for somehow only building the source code and not checking
the test code.

mokagio committed:
Make SwiftLint build script robust against spaces in paths

Gio the old man shakes fits at the sky and at spaces in paths

mokagio committed:
Convert some mock data from unnecessarily `NSLocalizedString` to `String`

mokagio committed:
Update `localized_comment` rule to not raise `"""` as false positive

mokagio committed:
Address remaining `localized_comment` violations

mokagio committed:
Revert making a delegate weak in the tests

mokagio committed:
Add a required `import` that somehow didn't block the debug build
